### PR TITLE
SDL: Use SDL_PeepEvents for event handling

### PR DESCRIPTION
**Pull request by dos1**
This should improve performance on platforms where SDL_PumpEvents
is expensive, like Emscripten.

### CODE REVIEW

**Comment by dos1** [comment]
Updated to get rid of C99ism.

File: src/sdl/sdl_system.c [hunk 1]
```diff
@@ -73,54 +75,58 @@ static char const *event_name(int type)
...
-                  _al_sdl_keyboard_event(&event);
-                  break;
-               case SDL_WINDOWEVENT_CLOSE:
-               case SDL_WINDOWEVENT_RESIZED:
-                  _al_sdl_display_event(&event);
-                  break;
-            }
+   SDL_PumpEvents();
+   int n = 0, i;
+   while ((n = SDL_PeepEvents(events, ALLEGRO_SDL_EVENT_QUEUE_SIZE, SDL_GETEVENT, SDL_FIRSTEVENT, SDL_LASTEVENT))) {
```
Comment (SiegeLord) [inline src/sdl/sdl_system.c:83]: This should be > 0, if I'm reading the documentation correctly.

Comment (dos1) [reply src/sdl/sdl_system.c:83]: Good catch! Updated.

**Comment by SiegeLord** [comment]
Thanks!

**Pull request merged**